src/components/result-catalog/__test__/ResultCatalog.test.tsx

```diff
@@ -0,0 +1,129 @@
+import { act, render, screen, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import ResultCatalog from '../ResultCatalog';
+import { BrowserRouter, LinkProps, Route, Routes } from 'react-router-dom';
+import { AppContext } from '../../../stores/SearchContext';
+import userEvent from '@testing-library/user-event';
+
+jest.mock('../../../service/apiService');
+
+describe('ResultCatalog', () => {
+  const props = {
+    startPage: false,
+    handleStopSearch: () => {},
+  };
+  const apiData = {
+    results: [
+      {
+        id: 1,
+        name: 'Morty Smith',
+        gender: 'Male',
+        species: 'Human',
+        status: 'Alive',
+        image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
+      },
+    ],
+    total: 1,
+  };
+  jest.fn().mockResolvedValue(apiData);
```

Comment:
This line has no sense. It makes sense only when you assign the mock to a variable and then use this variable, e.g.:
```
const mockGetApiData = jest.fn().mockResolvedValue(apiData);
console.log(mockGetApiData()) // => apiData
```

Reply:
will be fixed in the next PR